src/main/java/core/basesyntax/db/Storage.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.db;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private final Map<String, Integer> storage = new HashMap<>();
+
+    public void addFruit(String fruit, int quantity) {
+        storage.put(fruit, storage.getOrDefault(fruit, 0) + quantity);
+    }
+
+    public void removeFruit(String fruit, int quantity) {
+        int currentQuantity = storage.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Insufficient stock for " + fruit);
+        }
+        storage.put(fruit, currentQuantity - quantity);
+    }
+
+    public int getQuantity(String fruit) {
+        return storage.getOrDefault(fruit, 0);
+    }
+
+    public void updateFruit(String fruit, int quantity) {
+        storage.put(fruit, quantity);
+    }
+
+    public Map<String, Integer> getStorage() {
+        return Collections.unmodifiableMap(storage);
+    }
+
+    public void setFruitBalance(String fruit, int quantity) {
+        storage.put(fruit, quantity);
+    }
+}

```

Comment:
Ensure that the class `Storage` is used only for storing data and does not contain any business logic. Methods like `addFruit`, `removeFruit`, and `updateFruit` add some business logic to this class, which could be moved to a more appropriate place following the Single Responsibility Principle.